soh/soh/Enhancements/randomizer/hook_handlers.cpp

```diff
@@ -1591,6 +1591,17 @@ void RandomizerOnVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, va_l
             }
             break;
         }
+        case VB_GIVE_RANDO_GLITCH_FISHING_PRIZE: {
+            if (IS_RANDO) {
+                VBFishingData* fishData = va_arg(args, VBFishingData*);
```

Comment:
What you are passing into the _Should on the .c side is not a `VBFishingData*`, it is actually a `Fishing*`.

It happens to work currently for you because both of these structs have `Actor actor` as the first struct member, but using anything else would be Undefined behavior.
```suggestion
                Fishing* fishing = va_arg(args, Fishing*);
```